### ISSUE TITLE
Only the first visited https domain resolves correctly.

### ISSUE THREAD

**Issue by thenbe** [tailscale/caddy-tailscale#66]
```
https://foo.tailf123a.ts.net {
	bind tailscale/foo
	reverse_proxy localhost:3000
}

https://bar.tailf123a.ts.net {
	bind tailscale/bar
	reverse_proxy localhost:5000
}
```

Two domains are declared in the previous caddyfile. After starting caddy, only the first domain I visit will resolve correctly. The second one will never resolve. For example:

1. Start caddy.
1. Go to `https://foo.tailf123a.ts.net`
1. Caddy does its tls thing and I'm able to connect.
1. Now go to `https://bar.tailf123a.ts.net`.
1. Caddy this time tries to do a `http-01` challenge. This always fails. We are unable to connect `https://bar.tailf123a.ts.net`.

If we restart caddy, and visit `https://bar.tailf123a.ts.net` first, then it works. But then we can never conenct to `https://foo.tailf123a.ts.net` because that will do a http-01 challenge which fails.

If we declare 3 tailscale https domains in the caddyfile, the first one we visit will resolve correctly and the other 2 always fail.

**Comment by willnorris** [comment]
I'll work on testing and reproducing, but I'm curious what happens if you manually specify the tailscale cert manager for each site:

```
https://foo.tailf123a.ts.net {
	bind tailscale/foo
	tls {
		get_certificate tailscale
	}
	reverse_proxy localhost:3000
}

https://bar.tailf123a.ts.net {
	bind tailscale/bar
	tls {
		get_certificate tailscale
	}
	reverse_proxy localhost:5000
}
```

Since you have the full hostname you shouldn't actually need that, but given that it's falling back to an http-01 challenge, it seems like the tailscale cert manager isn't getting used.

**Comment by willnorris** [comment]
Okay, I can definitely reproduce and I see the same http-01 challenge for the second site.  It doesn't appear that manually setting the cert manager helps, which is weird.  The simplest complete config I've got to reproduce the error:

```caddyfile
{
  debug
  tailscale {
    ephemeral # create all nodes as ephemeral
  }
}

:443 {
  bind tailscale/caddytest
  tls {
    get_certificate tailscale
  }
  respond "Hello from caddytest"
}

:443 {
  bind tailscale/caddytest2
  tls {
    get_certificate tailscale
  }
  respond "Hello from caddytest2"
}
```

Debug output shows caddy trying to do http-01 then tls-apln-01 challenges.  This *looks* like a caddy bug, but I'm still digging into it.

/cc @mholt @mohammed90 as fyi

**Comment by thenbe** [comment]
Yup I get the same results you did when I specify the tailscale cert manager for each site.

Meanwhile, if anyone else encounters this issue they can use this workaround for now:

```
{
  auto_https off
}

:80 {
	bind tailscale/foo
	reverse_proxy localhost:3000
}

:443 {
	bind tailscale+tls/foo
	reverse_proxy localhost:3000
}

:80 {
	bind tailscale/bar
	reverse_proxy localhost:5000
}

:443 {
	bind tailscale+tls/bar
	reverse_proxy localhost:5000
}
```

**Comment by willnorris** [comment]
okay, I've found the problem.  While the [localAPIDialer](https://github.com/tailscale/caddy-tailscale/blob/5cc2140a0eb3b884de161e26d37cc4098de91d57/module.go#L324) I added does correctly find the right tsnet server to connect to, it's only ever dialed on first connection (tscert only [builds a single client](https://github.com/tailscale/tscert/blob/bbccfbf48933da858ca2f459342f2457be107029/tscert.go#L81-L87)).

We could possibly have localAPIDialer return a `net.Conn` that muxes reads and writes to the correct backend, but there's no context to mux on at that point.  So I'm not immediately sure how I'll fix this 😕
